data-access/mandataris-download.ts

```diff
@@ -0,0 +1,243 @@
+import { query, sparqlEscapeString } from 'mu';
+import { getSparqlResults } from '../util/sparql-result';
+import { sparqlEscapeDateTime, sparqlEscapeUri } from '../util/mu';
+
+export const downloadMandatarissen = {
+  getWithFilters,
+  getPropertiesOfMandatarissen,
+};
+
+async function getWithFilters(filters) {
+  const {
+    bestuursperiodeId,
+    bestuursorgaanId,
+    persoonIds,
+    fractieIds,
+    hasFilterOnOnafhankelijkeFractie,
+    hasFilterOnNietBeschikbareFractie,
+    bestuursFunctieCodeIds,
+  } = filters;
+  let bestuursorgaanInTijdFilter: string | null = null;
+  let onlyActiveFilter: string | null = null;
+  let persoonFilter: string | null = null;
+  let fractieFilter: string | null = null;
+  let fractieOnafhankelijkFilter: string | null = null;
+  let fractieOOBFilter: string | null = null;
+  let mandaatTypeFilter: string | null = null;
+
+  if (bestuursorgaanId) {
+    bestuursorgaanInTijdFilter = `
+      ?mandaat ^org:hasPost ?bestuursorgaanInTijd.
+      ?bestuursorgaanInTijd mu:uuid ${sparqlEscapeString(bestuursorgaanId)}.
+    `;
+  }
+  if (filters.onlyShowActive) {
+    const escapedTodaysDate = sparqlEscapeDateTime(new Date());
+    onlyActiveFilter = `
+      ?mandataris mandaat:einde ?einde.
+
+      ?bestuursorgaanInTijd mandaat:bindingStart ?startBestuursorgaan. 
+      ?bestuursorgaanInTijd mandaat:bindingEinde ?eindeBestuursorgaan. 
+
+      FILTER (
+        ?einde <= ?eindeBestuursorgaan &&
+        ?einde >= ?startBestuursorgaan &&
+        ?einde >= ${escapedTodaysDate}
+      )
+    `;
+  }
+
+  if (persoonIds.length >= 1) {
+    const idValues = persoonIds.map((id) => sparqlEscapeString(id)).join(' ');
+    persoonFilter = `
+      VALUES ?persoonId { ${idValues} }
+
+      ?mandataris mandaat:isBestuurlijkeAliasVan ?persoon.
+      ?persoon mu:uuid ?persoonId.
+    `;
+  }
+
+  if (fractieIds.length >= 1) {
```

Comment:
would extract the whole fractions filter to its own method and build a union clause there (possibly with only one segment if there is only one case).

I think it will make it more clear what is happening